src/app/oauth2/middleware.js

```diff
@@ -3,47 +3,72 @@ const { logCoreBackCall, transformError } = require("../shared/loggerHelper");
 const { LOG_COMMUNICATION_TYPE_REQUEST } = require("../shared/loggerConstants");
 const { getIpAddress } = require("../shared/ipAddressHelper");
 const coreBackService = require("../../services/coreBackService");
+const {
+  handleBackendResponse,
+  checkForIpvAndOauthSessionId,
+  journeyApi,
+} = require("../ipv/middleware");
 
-module.exports = {
-  setIpAddress: (req, res, next) => {
-    req.session.ipAddress = getIpAddress(req);
-    next();
-  },
-
-  setIpvSessionId: async (req, res, next) => {
-    try {
-      const authParams = {
-        responseType: req.query.response_type,
-        clientId: req.query.client_id,
-        redirectUri: req.query.redirect_uri,
-        state: req.query.state,
-        scope: req.query.scope,
-        request: req.query.request,
-      };
-
-      if (!authParams.request) {
-        return next(new Error("Request JWT Missing"));
-      }
-      if (!authParams.clientId) {
-        return next(new Error("Client ID Missing"));
-      }
-
-      logCoreBackCall(req, {
-        logCommunicationType: LOG_COMMUNICATION_TYPE_REQUEST,
-        path: API_SESSION_INITIALISE,
-      });
-
-      const response = await coreBackService.postSessionInitialise(
-        req,
-        authParams,
-      );
-
-      req.session.ipvSessionId = response?.data?.ipvSessionId;
-    } catch (error) {
-      transformError(error, `error handling journey page: ${req.params}`);
-      return next(error);
+function setIpAddress(req, res, next) {
+  req.session.ipAddress = getIpAddress(req);
+  next();
+}
+
+async function setIpvSessionId(req, res, next) {
+  try {
+    const authParams = {
+      responseType: req.query.response_type,
+      clientId: req.query.client_id,
+      redirectUri: req.query.redirect_uri,
+      state: req.query.state,
+      scope: req.query.scope,
+      request: req.query.request,
+    };
+
+    if (!authParams.request) {
+      return next(new Error("Request JWT Missing"));
     }
+    if (!authParams.clientId) {
+      return next(new Error("Client ID Missing"));
+    }
+
+    logCoreBackCall(req, {
+      logCommunicationType: LOG_COMMUNICATION_TYPE_REQUEST,
+      path: API_SESSION_INITIALISE,
+    });
+
+    const response = await coreBackService.postSessionInitialise(
+      req,
+      authParams,
+    );
 
-    next();
-  },
+    req.session.ipvSessionId = response?.data?.ipvSessionId;
+  } catch (error) {
+    transformError(error, `error handling journey page: ${req.params}`);
+    return next(error);
+  }
+
+  next();
+}
+
+async function handleOAuthJourneyAction(req, res, next) {
+  try {
+    checkForIpvAndOauthSessionId(req, res);
+    await handleOAuthResponse(req, res, "next");
+  } catch (error) {
+    transformError(error, "error invoking handleOAuthJourneyAction");
+    next(error);
```

Comment:
I think having a separate handler here is fine, but I wonder if we should still be using the same `handleJourneyResponse` function?

I think ideally we'd make the `pageId` an optional parameter to `handleJourneyResponse` and pass it explicitly from callers (tying into the comment above - we can then pass it explicitly, rather than deconstructing the request URL)